datafusion/core/src/sql/planner.rs

```diff
@@ -889,9 +889,10 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
 
                 // remove join expressions from filter
                 match remove_join_expressions(&filter_expr, &all_join_keys)? {
-                    Some(filter_expr) => {
-                        LogicalPlanBuilder::from(left).filter(filter_expr)?.build()
-                    }
+                    Some(filter_expr) => Ok(LogicalPlan::Filter(Filter {
+                        predicate: filter_expr,
+                        input: Arc::new(left),
+                    })),
```

Comment:
This is the bug fix. We cannot delegate to the `LogicalPlanBuilder` to build this `Filter` expression because it is not subquery-aware.